protocols/client/client_test.go

```diff
@@ -132,6 +136,7 @@ func TestNewAgentClient(t *testing.T) {
 	// wait mock server to stop
 	mock.Stop()
 	<-waitCh
+	os.Remove(mockSockAddr)
```

Comment:
Nit: This call (and the one below) could be put into a `defer` and moved to just below the error check after calling `startMockServer()`.

Reply:
Fixed, thank you.